src/lib-components/forms/DateRangePicker.vue

```diff
@@ -51,7 +53,34 @@ onMounted(() => {
         })
       }
     },
-  })
+  }
+
+  if (props.allowedRangeInDays) {
+    // Set the range to a prefilled value given the allowed range
+    const daysAgo = new Date()
+    opts.defaultDate = [
+      daysAgo.setDate(daysAgo.getDate() - props.allowedRangeInDays),
+      new Date(),
+    ]
+
+    // Handle onChange to dynamically adjust maxDate to x days ahead of the selected start date
+    opts.onChange = (selectedDates, _, self) => {
+      if (selectedDates.length === 1) {
+        // Clone date so as to not change selectedDates[0] value
+        var daysAhead = new Date(selectedDates[0].getTime())
+        daysAhead.setDate(daysAhead.getDate() + props.allowedRangeInDays)
+        const now = new Date()
+
+        if (daysAhead > now) {
+          daysAhead = now
+        }
+
+        self.set("maxDate", daysAhead)
```

Comment:
I think you'll need to set the `minDate` here is well to prevent backward range selections outside the range bounds.